docs/spec/lightclient/detection/detection.md

```diff
@@ -1,19 +1,199 @@
 ***This an unfinished draft. Comments are welcome!***
 
-## TODOs
+This document contains:
+
+- the outcome of recent discussion
+- a sketch of the light client supervisor to provide the context in
+  which fork detection happens
+- a discussion about lightstore semantics
+- a draft of the light node fork detection including "proof of fork"
+  definition, that is, the data structure to submit evidence to full
+  nodes.
+  
 
-- the main logic, i.e., describing what happens when the function of
-  the current spec returns. If a fork is detected, we need to report
-  evidence (and shut down light client?). + pre and post conditions on
-  lightstore when `VerifyToTarget` or detector are called.
+## Results of Discussions and Decisions 
+
+- Generating a minimal proof of fork (as suggested in [Issue #5083](https://github.com/tendermint/tendermint/issues/5083)) is too costly at the light client
+    - we do not know all lightblocks from the primary
+	- therefore there are many scenarios. we might even need to ask
+      the primary again for additional lightblocks to isolate the
+      branch. 
+> For instance, the light node starts with block at height 1 and the
+> primary provides a block of height 10 that the light node can
+> verify immediately. In cross-checking, a secondary now provides a
+> conflicting header b10 of height 10 that needs another header b5
+> of height 5 to
+> verify. Now, in order for the light node to convince the primary:
+>   - The light node cannot just sent b5, as it is not clear whether
+>     the fork happened before or after 5
+>   - The light node cannot just send b10, as the primary would also 
+>     need  b5 for verification
+>   - In order to minimize the evidence, the light node may try to
+>     figure out where the branch happens, e.g., by asking the primary
+>     for height 5 (it might be that more queries are required, also
+>     to the secondary. However, assuming that in this scenario the
+>     primary is faulty it may not respond.
+	  
+	  As the main goal is to catch misbehavior of the primary,
+      evidence generation and punishment must not depend on their
+      cooperation. So the moment we have proof of fork (even if it
+      contains several light blocks) we should submit right away.
+
+
+- decision: "full" proof of fork consists of two traces that origin in the
+  same lightblock and lead to conflicting headers of the same height.
   
-- We should clarify what is the expectation of VerifyToTarget so if it
-  returns TimeoutError it can be assumed faulty. I guess that
-  VerifyToTarget with correct full node should never terminate with
-  TimeoutError.
+- For submission of proof of fork, we may do some optimizations, for
+  instance, we might just submit  a trace of lightblocks that verifies a block
+  different from the one the full node knows (we do not send the trace
+  the primary gave us back to the primary)
+
+- The light client attack is via the primary. Thus we try to
```

Comment:
As said above, it may only lead to a safety issue, if it comes from the primary, and we gave the verified header to the use. If the verified header that comes from the primary is OK, the secondary may just force the lightclient to shut down and report evidence, but it will not make the light client have a bad verified block.

Reply:
The key property detection should ensure is the following: if a bad light block is "consumed" (either verified in the light node case or installed on chain in IBC case), it will be detected within bounded time. In case of the light node, this is ensured by checking verified block against witnesses, while in IBC case it done by a correct relayer checking every installed header. This ensures that exposed misbehaviour is always slashable which incentivise correct behaviour. In the scenario with correct primary and fork coming from the secondary, we will as Josef pointed out, also submit fork and shut down, but this scenario is less severe as "consumed" block is from the correct branch.